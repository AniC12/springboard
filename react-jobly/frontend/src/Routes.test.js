import '@testing-library/jest-dom';
import React from "react";
import { render, screen } from "@testing-library/react";
import Routes from "./Routes";
import { MemoryRouter } from "react-router";
import { UserProvider } from "./testUtils";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("renders Homepage for the root route", () => {
    render(
        <MemoryRouter initialEntries={["/"]}>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(screen.getByText(/Welcome Back/)).toBeInTheDocument(); // Adjust based on actual content
});

it("renders LoginForm for /login", () => {
    render(
        <MemoryRouter initialEntries={["/login"]}>
            <Routes />
        </MemoryRouter>
    );

    expect(screen.getByText("Log In")).toBeInTheDocument();
});