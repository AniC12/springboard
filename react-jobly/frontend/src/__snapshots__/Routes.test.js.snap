// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div>
      <h1>
        Jobly
      </h1>
      <p>
        All the jobs in one, convenient place.
      </p>
      <h2>
        Welcome Back, testuser!
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
